ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/TransactionServiceOutputsIT.scala

```diff
@@ -56,4 +61,77 @@ class TransactionServiceOutputsIT extends LedgerTestSuite {
       assertEquals("AgreementTextDefault", contract.getAgreementText, "")
     }
   })
+
+  test(
+    "TXVerbosity",
+    "Expose field names only if the verbose flag is set to true",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      _ <- ledger.create(party, Dummy(party))
+      request = ledger.getTransactionsRequest(Seq(party))
+      verboseTransactions <- ledger.flatTransactions(request.update(_.verbose := true))
+      verboseTransactionTrees <- ledger.transactionTrees(request.update(_.verbose := true))
+      nonVerboseTransactions <- ledger.flatTransactions(request.update(_.verbose := false))
+      nonVerboseTransactionTrees <- ledger.transactionTrees(request.update(_.verbose := false))
+    } yield {
+      assertLabelsAreExposedCorrectly(
+        party,
+        verboseTransactions,
+        verboseTransactionTrees,
+        labelIsNonEmpty = true,
+      )
+      assertLabelsAreExposedCorrectly(
+        party,
+        nonVerboseTransactions,
+        nonVerboseTransactionTrees,
+        labelIsNonEmpty = false,
+      )
+    }
+  })
+
+  private def assertLabelsAreExposedCorrectly(
+      party: Primitive.Party,
+      transactions: Seq[Transaction],
+      transactionTrees: Seq[TransactionTree],
+      labelIsNonEmpty: Boolean,
+  ): Unit = {
+    assert(transactions.nonEmpty, s"$party expected non empty transaction list")
+    assert(transactionTrees.nonEmpty, s"$party expected non empty transaction tree list")
+    val createArgs = flatTransactionFields(transactions)
+    val text = labelIsNonEmpty match {
+      case true => "with"
+      case false => "without"
+    }
+    assert(
+      createArgs.forall(_.label.nonEmpty == labelIsNonEmpty),
+      s"$party expected a contract $text labels, but received $transactions.",
+    )
+    val treeCreateArgs = transactionFields(transactionTreeEvents(transactionTrees))
+    assert(
+      treeCreateArgs.forall(_.label.nonEmpty == labelIsNonEmpty),
+      s"$party expected a contract $text labels, but received $transactionTrees.",
+    )
+  }
+
+  private def transactionTreeEvents(transactionTrees: Seq[TransactionTree]): Seq[CreatedEvent] = {
+    for {
+      transactionTree <- transactionTrees
+      eventById <- transactionTree.eventsById
+      (_, tree) = eventById
+      events = tree
+      created = events.getCreated
+    } yield created
+  }
+
+  private def flatTransactionFields(transactions: Seq[Transaction]): Seq[RecordField] =
```

Comment:
This function looks like it crashes if there is an archive in the transaction. That’s fine for your test but less fine for a general utility so maybe consider moving this within the test/inline it.